### PR TITLE
e2e tests with chainsimulator

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
Added e2e tests with chain simulator:
- a test that issues a contract, calls some of its function that emit logs and then the event processor should receive all the events
- a test that issues a token, transfers it to some addresses and then the event processor should receive all the events

Furthermore, there were some changes regarding the event processor:
- fix the situation when the identifiers are missing that could have caused an invalid es request
- add the option to filter by shard id (useful for esdt transfers when logs are emitted on both source and destination shard)

### CODE REVIEW

File: e2e-tests/chain-simulator-e2e/event.processor.e2e.spec.ts [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+      logMessage(`Transferred 5 tokens to Carol. Tx hash: ${carolTransferTxHash}}`);
+    }
+
+    await new Promise(resolve => setTimeout(resolve, 2000));
+
+    let numOfEventsReceived = 0;
+    let lastProcessedTimestamp = 0;
+    let counter = 0;
+    while (true) {
+      await eventProcessor.start(new EventProcessorOptions({
```
Comment (cfaur09) [inline e2e-tests/chain-simulator-e2e/event.processor.e2e.spec.ts:157]: * you have `eventProcessor.start` witch is used multiple time, maybe you can create a function to abstract the logic and that just call the function inside the testcase

File: e2e-tests/chain-simulator-e2e/event.processor.e2e.spec.ts [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+      deployer: ALICE_ADDRESS,
+      contractCodeRaw: contractCodeRaw,
+      hexArguments: contractArgs,
+    }));
+    logMessage(`Deployed ping pong SC. Address: ${scAddress}`);
+
+    const numPingPongs = 20;
+
+    for (let i = 0; i < numPingPongs; i++) {
+      const pingTxHash = await sendTransaction({
```
Comment (cfaur09) [inline e2e-tests/chain-simulator-e2e/event.processor.e2e.spec.ts:67]: * same for the sendTransaction, maybe it's more robust to create a function for sendTransaction

File: e2e-tests/chain-simulator-e2e/event.processor.e2e.spec.ts [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+      logMessage(`Running event processor #${counter + 1}`);
+
+      counter++;
+      if (counter > 10 || numOfEventsReceived === 2 * numPingPongs) {
+        break;
+      }
+    }
+
+    expect(numOfEventsReceived).toBe(numPingPongs * 2);
+  }, 100000);
```
Comment (cfaur09) [inline e2e-tests/chain-simulator-e2e/event.processor.e2e.spec.ts:120]: * this timeout value is not to big ?

Comment (bogdan-rosianu) [reply e2e-tests/chain-simulator-e2e/event.processor.e2e.spec.ts:120]: it means 1min40s, so it's fine

File: e2e-tests/chain-simulator-e2e/event.processor.e2e.spec.ts (hunk 1 above)
Comment (bogdan-rosianu) [reply e2e-tests/chain-simulator-e2e/event.processor.e2e.spec.ts:157]: well, i call it with different arguments each time. I don't see a benefit extracting a function that just calls a function

File: e2e-tests/chain-simulator-e2e/event.processor.e2e.spec.ts (hunk 2 above)
Comment (bogdan-rosianu) [reply e2e-tests/chain-simulator-e2e/event.processor.e2e.spec.ts:67]: well, i call it with different arguments each time. I don't see a benefit extracting a function that just calls a function

File: e2e-tests/chain-simulator-e2e/event.processor.e2e.spec.ts (hunk 3 above)
Comment (cfaur09) [reply e2e-tests/chain-simulator-e2e/event.processor.e2e.spec.ts:120]: okay, just to be sure that the time is ok

**Review by cfaur09 — APPROVED** [review]

**Pull request merged**